discovery_engine_core/core/src/stack/data.rs

```diff
@@ -41,7 +41,7 @@ pub(crate) struct Data {
     #[derivative(Default(value = "1."))]
     pub(super) beta: f32,
     /// Documents in the [`Stack`](super::Stack).
-    pub(super) documents: Vec<Document>,
+    pub(crate) documents: Vec<Document>,
```

Comment:
:seedling: The abstraction of stack is currently broken as all stack internal logic besides getting new documents is now in `update_stacks` which is in `engine.rs`. I propose to move `update_stacks` into `stacks.rs` in some follow up PR and then set it back to `pub(super)`. Also we can remove `Ops::merge` as by now we always drain the stack documents and then set a new list of stack documents instead.